### ISSUE TITLE
What are the correct steps for TensorflowSharp using the GPU?

### ISSUE THREAD

**Issue by bzburr** [migueldeicaza/TensorFlowSharp#290]
HI there

I have a working CPU based TensorflowSharp object detection app. *happy days* nit at 1.5 seconds per images and thousands of images to process i need to speed it up :)

I also have a working model that is built using the GPU on the same PC.

Can someone tell me exactly what i need to do to make my TensorflowSharp use the GPU.

I've tried downloading the  libtensorflow_gpu.dll etc and copying them to the suggested folders, i've tried using the _python_pywrap_tensorflow_internal.pyd renamed to libtensorflow...- but the gpu is isn't getting hit - is there something i need to do when creating the TFSession object?

What might have i missed?

**Comment by edwardyehuang** [comment]
I don't think your Tensorflowsharp is using libtensorflow_gpu.dll. Maybe you need to rename it to libtensorflow.dll. Also make sure your libtensorflow.dll is larger than 200mb (If you are using 1.7)

**Comment by bzburr** [comment]
yes i shall write it up and post a "how to" for future reference :)

**Comment by migueldeicaza** [comment]
You do need to replace the libraries I distribute with the GPU versions.

It is on my TO-Do list to package both and swtich those at runtime.   I am just in deep agony at .NET still lacking support for doing this swapping.

**Comment by MykolaKovalchuk** [comment]
Don't want to start one more thread about GPU...

Can someone help me, where I can find libtensorflow.so and libtensorflow_framework.so in Linux (Ubuntu), which are used by Python, and which I have built to work with GPU?

I have found libtensorflow_framework.so in one of Python subfolders, but I cannot find libtensorflow.so anywhere. And when I replace libtensorflow_framework.so in TensorFlowSharp/runtimes subfolder, it does not work with error: "cannot find dll libtensorflow".

**Comment by MykolaKovalchuk** [comment]
Btw, assemblies in .Net can be loaded manually. So I would have several classes in different assemblies with methods bound to different tensorflow libraries (CPU, GPU), and in runtime initialize class from corresponding assembly.

**Comment by MykolaKovalchuk** [comment]
I have downloaded pre-built libtensorflow.so and libtensorflow_framework.so from tensorflow site as described here: https://www.tensorflow.org/install/install_c
Seems to be working.
